pkg/controller/jobframework/interface.go

```diff
@@ -44,7 +43,7 @@ type GenericJob interface {
 	// Finished means whether the job is completed/failed or not,
 	// condition represents the workload finished condition.
 	// Observed generation of the workload is set by the jobframework.
-	Finished() (condition metav1.Condition, finished bool)
+	Finished() (reason, message string, finished bool)
```

Comment:
Nit: How about instead of `reason` we use a `success` bool and pick the reason in the jobframework.

Reply:
+1. That would leave less room for custom implementation to return arbitrary strings that are hard to standardize.